test/e2e/nodepool_create_test.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	api "github.com/Azure/ARO-HCP/internal/api/v20240610preview/generated"
+	"github.com/Azure/ARO-HCP/test/util/labels"
+)
+
+var _ = Describe("Put HCPOpenShiftCluster Nodepool", func() {
+	var (
+		NodePoolsClient *api.NodePoolsClient
+	)
+
+	BeforeEach(func() {
+		By("Preparing HCP nodepools client")
+		NodePoolsClient = clients.NewNodePoolsClient()
+	})
+
+	It("Attempts to create a nodepool for a non-existant HCPOpenshiftCluster", labels.Medium, labels.Negative, func(ctx context.Context) {
+		var (
+			nodePoolName     = "mynodepool"
+			clusterName      = "non-existing_cluster"
+			nodePoolResource api.NodePool
+			nodePoolOptions  *api.HcpOpenShiftClustersClientBeginCreateOrUpdateOptions
```

Comment:
Why not type this as `*api.NodePoolsClientBeginCreateOrUpdateOptions` and not need to cast? You only ever use it once, where does this type come from?

Reply:
I think when I first started writing this there wasn't a NodePoolsClientBeginCreateOrUpdateOptions yet. Thanks for catching this, I've changed nodePoolOptions to use NodePoolsClientBeginCreateOrUpdateOptions.